component/storage/edv/encryptedformatter.go

```diff
@@ -118,52 +168,72 @@ func (e *EncryptedFormatter) getStructuredDocFromEncryptedDoc(
 	return structuredDoc, nil
 }
 
-// TODO (#2376) Revisit how we're generating EDV document IDs, since it's technically not 100% in line with the spec.
-//  (Spec requires randomly generated IDs)
 // Generates an encrypted document ID based off of key.
-func (e *EncryptedFormatter) formatKey(key string) (string, error) {
-	keyHash, err := e.macCrypto.ComputeMAC([]byte(key))
+func (e *EncryptedFormatter) generateDocumentIDDeterministically(prefix, key string) (string, error) {
+	if key == "" {
+		return "", nil
+	}
+
+	keyHash, err := e.macCrypto.ComputeMAC([]byte(prefix + key))
 	if err != nil {
 		return "", fmt.Errorf(`failed to compute MAC based on key "%s": %w`, key, err)
 	}
 
 	return base58.Encode(keyHash[0:16]), nil
 }
 
-func (e *EncryptedFormatter) formatTags(tagPrefix string, tags []spi.Tag) ([]spi.Tag, error) {
+func generateDocumentIDRandomly() (string, error) {
```

Comment:
consider update naming to "generateRandomDocumentID"

Reply:
I agree, that's shorter and better. Done!